src/libPOW/pow.cpp

```diff
@@ -238,8 +242,250 @@ ethash_mining_result_t POW::MineFullGPU(uint64_t blockNum,
   return ethash_mining_result_t{"", "", 0, false};
 }
 
-void POW::MineFullGPUThread(uint64_t blockNum,
-                            ethash_hash256 const& header_hash,
+ethash_mining_result_t POW::RemoteMine(const PairOfKey& pairOfKey,
+                                       uint64_t blockNum,
+                                       ethash_hash256 const& headerHash,
+                                       ethash_hash256 const& boundary) {
+  LOG_MARKER();
+
+  m_shouldMine = true;
+
+  ethash_mining_result_t miningResult{"", "", 0, false};
+  if (!SendWorkToProxy(pairOfKey, blockNum, headerHash, boundary)) {
+    LOG_GENERAL(WARNING, "Failed to send work package to mining proxy.");
+    return miningResult;
+  }
+
+  bool verifyResult = CheckMiningResult(pairOfKey, blockNum, headerHash,
+                                        boundary, miningResult);
+  if (!verifyResult) {
```

Comment:
return from here??

Reply:
No, if verify fail, also need to send the result to Proxy, let proxy know it is a invalid solution.